src/bootstrap/format.rs

```diff
@@ -193,11 +193,50 @@ pub fn format(build: &Builder<'_>, check: bool, paths: &[PathBuf]) {
     let (tx, rx): (SyncSender<PathBuf>, _) = std::sync::mpsc::sync_channel(128);
     let walker = match paths.get(0) {
         Some(first) => {
-            let mut walker = WalkBuilder::new(first);
-            for path in &paths[1..] {
-                walker.add(path);
+            let find_shortcut_candidates = |first_file_name| {
+                let mut candidates = Vec::new();
+                for candidate in WalkBuilder::new(src.clone()).max_depth(Some(3)).build() {
+                    if let Ok(entry) = candidate {
+                        if let Ok(meta) = entry.metadata() {
+                            if meta.is_dir() && entry.file_name() == first_file_name {
+                                candidates.push(entry.into_path());
+                            }
+                        }
+                    }
+                }
+                candidates
+            };
+
+            let first_file_name = first.file_name().unwrap_or_else(|| {
+                eprintln!("failed to convert argument(s) into file name");
+                crate::detail_exit(1);
+            });
+
+            if let [single_candidate] = &find_shortcut_candidates(first_file_name)[..] {
```

Comment:
This seems wrong. This will always start finding the related directory from the last component of the requested path even if we provide it.

An example, if I execute `x fmt library/std`, it will take `std` part and try to find `library`.

Also, another problem here is that if I execute `x fmt some/random/std` command, it will still work. `No such file or directory` error will never occur unless the last component of the requested path is not exists in source directory.

I think we should trigger this directory finding flow only when we provide single component paths from the `fmt` command.